webknossos-datastore/app/com/scalableminds/webknossos/datastore/datavault/VaultPath.scala

```diff
@@ -4,78 +4,34 @@ import com.scalableminds.util.io.ZipIO
 import net.liftweb.util.Helpers.tryo
 
 import java.net.URI
-import java.nio.file.{FileSystem, LinkOption, Path, Paths, WatchEvent, WatchKey, WatchService}
 import scala.collection.immutable.NumericRange
 
-/*
-VaultPath implements Path so that a drop in replacement is possible while continuing to use Paths for local storage.
-This class does not implement all relevant methods and it might be a good idea to remove the inheritance on Path in the
-future.
- */
-
-class VaultPath(uri: URI, dataVault: DataVault) extends Path {
-
-  protected def readBytesGet(range: Option[NumericRange[Long]]): Array[Byte] =
-    dataVault.readBytes(this, range)
+class VaultPath(uri: URI, dataVault: DataVault) {
 
   def readBytes(range: Option[NumericRange[Long]] = None): Option[Array[Byte]] =
-    tryo(readBytesGet(range)).toOption.map(ZipIO.tryGunzip)
-
-  override def getFileSystem: FileSystem = ???
-
-  override def isAbsolute: Boolean = ???
-
-  override def getRoot: Path = ???
+    tryo(dataVault.readBytes(this, range)).toOption.map(ZipIO.tryGunzip)
 
-  override def getFileName: Path =
-    Paths.get(uri.toString.split("/").last)
+  def fileName: String =
```

Comment:
I'm not sure if fileName is the best name here. Now that it is no longer forced upon us, we might want something else, since this is not always the fileName, but also the name of a directory. Possible suggestions:
- last uri element
- slug
- resource name
- basename (`man basename`)